rusk/src/lib/node/vm/config/opt.rs

```diff
@@ -0,0 +1,164 @@
+pub struct OptionalConfig {
+    /// The amount of gas points charged for each blob in a transaction
+    pub gas_per_blob: Option<u64>,
+
+    /// The amount of gas points charged for each byte in a contract-deployment
+    /// bytecode.
+    pub gas_per_deploy_byte: Option<u64>,
+
+    /// The minimum gas points charged for a contract deployment.
+    pub min_deploy_points: Option<u64>,
+
+    /// The minimum gas price set for a contract deployment
+    pub min_deployment_gas_price: Option<u64>,
+
+    /// The maximum amount of gas points that can be used in a block.
+    pub block_gas_limit: Option<u64>,
+
+    /// The timeout for a candidate block generation.
+    #[serde(default, with = "humantime_serde")]
+    pub generation_timeout: Option<Duration>,
+
+    /// Set of features to activate
+    #[serde(default)]
+    features: HashMap<String, u64>,
+}
+
+impl OptionalConfig {
+    pub fn feature(&self, feature: &str) -> Option<u64> {
+        self.features
+            .iter()
+            .find(|(k, _)| k.eq_ignore_ascii_case(feature))
+            .map(|(_, &v)| v)
+    }
+
+    pub fn with_feature<S: Into<String>>(
+        &mut self,
+        feature: S,
+        activation: u64,
+    ) {
+        let feature: String = feature.into();
+        // Check for case insensitive key
+        if self.feature(&feature).is_some() {
+            self.features.remove(&feature);
+        }
+        self.features.insert(feature, activation);
+    }
+
+    /// Injects the well-known configuration into this optional configuration.
+    ///
+    /// If a field is already set, a warning is logged and the existing value is
+    /// kept.
+    /// If a feature is already set, a warning is logged and the existing value
+    /// is kept.
+    /// If a feature is not recognized in the well-known configuration, a
+    /// warning is logged.
+    pub fn inject_network_conf(&mut self, config: WellKnownConfig) {
+        Self::set_or_warn(
+            "gas_per_blob",
+            &mut self.gas_per_blob,
+            config.gas_per_blob,
+        );
+        Self::set_or_warn(
+            "gas_per_deploy_byte",
+            &mut self.gas_per_deploy_byte,
+            config.gas_per_deploy_byte,
+        );
+        Self::set_or_warn(
+            "min_deploy_points",
+            &mut self.min_deploy_points,
+            config.min_deploy_points,
+        );
+        Self::set_or_warn(
+            "min_deployment_gas_price",
+            &mut self.min_deployment_gas_price,
+            config.min_deployment_gas_price,
+        );
+
+        Self::set_or_warn(
+            "block_gas_limit",
+            &mut self.block_gas_limit,
+            config.block_gas_limit,
+        );
+
+        for (feature, activation) in &config.features {
+            if let Some(v) = self.feature(feature) {
+                if v != *activation {
+                    warn!("[vm].feature {feature} is set to {v} (overriding the well-known network config value of {activation})");
```

Comment:
the log seems to state it's overriding the current value, but then it doesn't.
what's the intended behavior?